flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/common/CommonPhysicalTableSourceScan.scala

```diff
@@ -51,9 +52,21 @@ abstract class CommonPhysicalTableSourceScan(
   }
 
   override def explainTerms(pw: RelWriter): RelWriter = {
+    val version = extractSnapshotVersion()
     super
       .explainTerms(pw)
       .item("fields", getRowType.getFieldNames.asScala.mkString(", "))
       .itemIf("hints", RelExplainUtil.hintsToString(getHints), !getHints.isEmpty)
+      .itemIf("version", version.get, version.isDefined)
+  }
+
+  private def extractSnapshotVersion(): Option[String] = {
+    val originTable: CatalogBaseTable =
+      relOptTable.contextResolvedTable.getTable.asInstanceOf[CatalogBaseTable]
+    originTable match {
+      case catalogTable: CatalogTable if catalogTable.getSnapshot.isPresent =>
+        Option(catalogTable.getSnapshot.get().toString)
+      case _ => Option("")
```

Comment:
Because the term `version.get` will be executed even though `version` is empty during `explainTerms` ( see `.itemIf("version", version.get, version.isDefined)`), and use `version.getOrElse("")` doesn’t seem much difference compared to the current approach, WDYT?